app/upgrades/v2/upgrades.go

```diff
@@ -1,20 +1,177 @@
+
+			var keyTable paramstypes.KeyTable
+			switch subspace.Name() {
+			case evmtypes.ModuleName:
+				keyTable = evmtypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			case assettypes.ModuleName:
+				keyTable = assettypes.ParamKeyTable()
+			case bridgetypes.ModuleName:
+				keyTable = bridgetypes.ParamKeyTable()
+			case banktypes.ModuleName:
+				keyTable = banktypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			case stakingtypes.ModuleName:
+				keyTable = stakingtypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			case minttypes.ModuleName:
+				keyTable = minttypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			case distrtypes.ModuleName:
+				keyTable = distrtypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			case slashingtypes.ModuleName:
+				keyTable = slashingtypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			case govtypes.ModuleName:
+				keyTable = govv1.ParamKeyTable() //nolint: staticcheck // SA1019
+			case crisistypes.ModuleName:
+				keyTable = crisistypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			case authtypes.ModuleName:
+				keyTable = authtypes.ParamKeyTable() //nolint: staticcheck // SA1019
+			}
+
+			if !subspace.HasKeyTable() {
+				subspace.WithKeyTable(keyTable)
+			}
+		}
+
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+		legacyBaseAppSubspace := paramsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+		err := baseapp.MigrateParams(sdkCtx, legacyBaseAppSubspace, consensusKeeper.ParamsStore)
+		if err != nil {
+			return nil, err
+		}
+
+		legacyClientSubspace, _ := paramsKeeper.GetSubspace(exported.ModuleName)
+		var params clienttypes.Params
+		legacyClientSubspace.GetParamSet(sdkCtx, &params)
+
+		params.AllowedClients = append(params.AllowedClients, exported.Localhost)
+		ibcKeeper.ClientKeeper.SetParams(sdkCtx, params)
+
+		deleteKVStore(sdkCtx.KVStore(evmStoreKey))
+		delete(vm, evmtypes.ModuleName)
+		MigrateEthAccountsToBaseAccounts(sdkCtx, accountKeeper, evmKeeper)
+
+		// Run migrations and init genesis for bridge module
+		newVM, err := mm.RunMigrations(ctx, configurator, vm)
+		if err != nil {
+			return nil, err
+		}
+
+		// Update bridge genesis state
+		err = bridgeKeeper.Params.Set(ctx, bridgetypes.NewParams("realio15md2mg7w62xf53gdnv7m06lpumunuhqrm5fuxl"))
+		if err != nil {
+			return nil, err
+		}
+		err = bridgeKeeper.RegisteredCoins.Set(ctx, "ario", bridgetypes.RateLimit{
+			Ratelimit:     math.Int(math.NewUintFromString("1000000000000000000000000")),
+			CurrentInflow: math.ZeroInt(),
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = bridgeKeeper.RegisteredCoins.Set(ctx, "arst", bridgetypes.RateLimit{
+			Ratelimit:     math.Int(math.NewUintFromString("1000000000000000000000000")),
+			CurrentInflow: math.ZeroInt(),
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = bridgeKeeper.EpochInfo.Set(ctx, bridgetypes.EpochInfo{
+			StartTime:            time.Unix(int64(1729763876), 0),
```

Comment:
StartTime should not be fixed

Reply:
The start time is only set once in order to trigger the epoch counting. So it should be fixed base on the upgrade time.